connector/avro/src/main/scala/org/apache/spark/sql/avro/SchemaConverters.scala

```diff
@@ -51,11 +51,14 @@ object SchemaConverters {
    *
    * @since 2.4.0
    */
+  def toSqlType(avroSchema: Schema, useStableIdForUnionType: Boolean): SchemaType = {
+    toSqlTypeHelper(avroSchema, Set.empty, useStableIdForUnionType)
+  }
   def toSqlType(avroSchema: Schema): SchemaType = {
-    toSqlTypeHelper(avroSchema, Set.empty, AvroOptions(Map()))
+    toSqlTypeHelper(avroSchema, Set.empty, AvroOptions(Map()).useStableIdForUnionType)
   }
   def toSqlType(avroSchema: Schema, options: Map[String, String]): SchemaType = {
-    toSqlTypeHelper(avroSchema, Set.empty, AvroOptions(options))
```

Comment:
so this is the place that caused perf issue before?

Reply:
Yes, from the flame graph. We can know that create a `AvroOptions` is very expensive.

Reply:
Do we still have callers for this `def toSqlType` variant?

Reply:
There is a method called `toCatalystType` that call this `def toSqlType variant`. Shall I also change that one?

Reply:
Oh, It seems that there is no caller for this `def toSqlType` any more. Only in test suite. Shall I remove this?

Reply:
yea let's remove